### PR TITLE
(FACT-1520) Various fixes for Freebsd facts

### PR DESCRIPTION
**Pull request by jmslagle**
This adds networking facts, and fixes a case where the wrong system serial was used.

### CODE REVIEW

File: lib/src/facts/freebsd/dmi_resolver.cc [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace facter { namespace facts { namespace freebsd {
         result.bios_version = kenv_lookup("smbios.bios.version");
         result.bios_release_date = kenv_lookup("smbios.bios.reldate");
         result.uuid = kenv_lookup("smbios.system.uuid");
-        result.serial_number = kenv_lookup("smbios.planar.serial");
```
Comment (MikaelSmith) [inline lib/src/facts/freebsd/dmi_resolver.cc]: Was this ever accurate?

**Comment by branan** [comment]
How much of the freebsd networking resolver is copied from openbsd? Do we need to do some refactoring here to clean that up? I'm happy to help guide you through that

File: lib/src/facts/freebsd/dmi_resolver.cc (hunk 1 above)
Comment (jmslagle) [inline lib/src/facts/freebsd/dmi_resolver.cc]: No.

Well kinda?

It's the Planar serial - likely the MB serial.  Almost certainly NOT what you want.

**Comment by jmslagle** [comment]
Almost or even all of it is from OpenBSD - unlike the other PR I need to submit to do memory and others which are largely different.

I can attempt to refactor this to move these two functions upstream to the bsd resolver, but you had indicated it's in use by other things.

Having FreeBSD include the openbsd file seemed just as bad.  But DRY.

**Comment by jmslagle** [comment]
I can lurk on Slack if we want to discuss.

**Comment by MikaelSmith** [comment]
@branan is refactoring something you planned to help with? If not, this is small enough I'm not too worried about merging it.

**Pull request merged**